cpp/ql/lib/semmle/code/cpp/controlflow/SSAUtils.qll

```diff
@@ -153,9 +153,11 @@ library class SSAHelper extends int {
    * Modern Compiler Implementation by Andrew Appel.
    */
   private predicate frontier_phi_node(StackVariable v, BasicBlock b) {
-    exists(BasicBlock x | dominanceFrontier(x, b) and ssa_defn_rec(v, x)) and
+    exists(BasicBlock x |
+      dominanceFrontier(x, pragma[only_bind_into](b)) and ssa_defn_rec(pragma[only_bind_into](v), x)
```

Comment:
Yeah, now the bad join is ruled out through the two pragmas on `v`, so that's good. But what's the intention behind the pragma on `b` here? That doesn't do anything. If you wanted to lock down the order further than just saying "don't start with a join on `v`", then you'd need e.g.:
```suggestion
      dominanceFrontier(x, b) and ssa_defn_rec(pragma[only_bind_into](v), pragma[only_bind_into](x))
```
as that would force the `ssa_defn_rec` first in the pipeline (and then the entire order would be forced from there).

Reply:
My intention was indeed to force the join order completely. As I wrote [above](https://github.com/github/codeql/pull/7434#discussion_r771337168), I thought the pragma on `b` would prevent this join order: `live_at_start_of_bb(v, b) -> dominanceFrontier(x, b) -> ssa_defn_rec(v, x)`.

Reply:
It does. But it doesn't prevent `dominanceFrontier(x, b) -> live_at_start_of_bb(v, b) -> ssa_defn_rec(v, x)`, which is equivalent.

Reply:
Oh. I did not think of that one >.<